etna/models/statsforecast.py

```diff
@@ -0,0 +1,767 @@
+import warnings
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Sequence
+from typing import Tuple
+from typing import Union
+
+import numpy as np
+import pandas as pd
+from statsforecast.models import AutoARIMA
+from statsforecast.models import AutoCES
+from statsforecast.models import AutoETS
+from statsforecast.models import AutoTheta
+
+from etna.distributions import BaseDistribution
+from etna.distributions import IntDistribution
+from etna.libs.statsforecast import ARIMA
+from etna.models.base import BaseAdapter
+from etna.models.base import NonPredictionIntervalContextIgnorantAbstractModel
+from etna.models.base import PredictionIntervalContextIgnorantAbstractModel
+from etna.models.mixins import NonPredictionIntervalContextIgnorantModelMixin
+from etna.models.mixins import PerSegmentModelMixin
+from etna.models.mixins import PredictionIntervalContextIgnorantModelMixin
+from etna.models.utils import determine_freq
+from etna.models.utils import determine_num_steps
+
+StatsForecastModel = Union[AutoCES, AutoARIMA, AutoTheta, AutoETS, ARIMA]
+
+
+class _StatsForecastBaseAdapter(BaseAdapter):
```

Comment:
I currently think that it is better to rewrite this using `statsforecast.StatsForecast` class to utilize multiprocessing and `ray` utilities.
Probably we can modify our own code to work better with multiprocessing/`ray`. But I think that it won't happen in a very close future.